examples/experimental/i3s-17-and-debug/tile-layer/tile-layer.js

```diff
@@ -81,6 +82,80 @@ export default class TileLayer extends Tile3DLayer {
       }
     );
   }
+
+  renderLayers() {
+    const {tileset3d, layerMap, colorsMap} = this.state;
+
+    const {obbColoringMode, showObb} = this.props;
+
+    if (!tileset3d) {
+      return null;
+    }
+
+    let layers = tileset3d.tiles
+      .map(tile => {
+        const layerCache = (layerMap[tile.id] = layerMap[tile.id] || {
+          tile
+        });
+        let {layer} = layerCache;
+
+        if (tile.selected) {
+          if (!layer) {
+            layer = this._getSubLayer(tile);
+          } else if (layerCache.needsUpdate) {
+            layer = this._getSubLayer(tile, layer);
+            layerCache.needsUpdate = false;
+          } else if (!layer.props.visible) {
+            layer = layer.clone({
+              visible: true
+            });
+          }
+        } else if (layer && layer.props.visible) {
+          layer = layer.clone({
+            visible: false
+          });
+        }
+
+        layerCache.layer = layer;
+        return layer;
+      })
+      .filter(Boolean);
+
+    if (showObb) {
+      const debugLayers = tileset3d.tiles
+        .map(tile => {
+          const layerCache = (layerMap[`obb-debug-${tile.id}`] = layerMap[
+            `obb-debug-${tile.id}`
+          ] || {
+            tile
+          });
+          let {layer} = layerCache;
+
+          if (tile.selected) {
+            if (!layer) {
+              layer = getObbLayer(tile, {coloredBy: obbColoringMode, colorsMap});
```

Comment:
I think we should render this layer in the application, not inside the tile layer. You should extract an array of all the tiles and use that as the instance data table to a SimpleMeshLayer. We should not need to modify this function.

Reply:
Moved this rendering to separate class.